app/main.py

```diff
@@ -1,8 +1,36 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.__class__.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    person_list = []
+    for item in people:
+        person = Person(item["name"], item["age"])
+        for key in item.keys():
+            if key == "wife":
+                person.wife = item["wife"]
+                person.husband = None
+            if key == "husband":
+                person.husband = item["husband"]
+                person.wife = None
+        person_list.append(person)
+
+    for person in person_list:
+        if person.wife is not None:
+            for i in range(len(person_list)):
```

Comment:
use `Person.people` dict to get rid of an extra cycle

Reply:
Sorry for the late reply. There were problems with electricity